web/hooks/pull_request.go

```diff
@@ -0,0 +1,198 @@
+	})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			wh.logger.Debugf("Ignoring pull request closed event for non-managed pull request #%d, in %s",
+				payload.GetPullRequest().GetNumber(), payload.GetRepo().GetFullName())
+		} else {
+			wh.logger.Errorf("Failed to get pull request from database: %v", err)
+		}
+		return
+	}
+
+	if err := wh.db.HandleMergingPR(pullRequest); err != nil {
+		wh.logger.Errorf("Failed to delete pull request from database %v", err)
+	}
+	wh.logger.Debugf("Pull request successfully closed for repository: %s", payload.GetRepo().GetFullName())
+}
+
+// TODO(Meling): I think it makes sense to have this function here, instead of in assignments/pull_request.go,
+// since we are creating the pull request in the scope of a pull request opened event.
+// If I were to later implement a function for creating a pull request from another context, e.g. upon task creation,
+// it would make no sense to use the following process.
+// But if you have a better idea, please let me know.
+
+// createPullRequest creates a new pull request record from a pull request opened event.
+func (wh GitHubWebHook) createPullRequest(payload *github.PullRequestEvent, repo *pb.Repository) {
+	wh.logger.Debugf("Attempting to create pull request for repository: %s", payload.GetRepo().GetFullName())
+	issueNumber, err := getLinkedIssue(payload.GetPullRequest().GetBody())
+	if err != nil {
+		wh.logger.Debugf("Failed to get issue number from pull request body: %v, in repository %s", err, payload.GetRepo().GetFullName())
+		return
+	}
+	var associatedIssue *pb.Issue = nil
+	for _, issue := range repo.Issues {
+		if issue.IssueNumber == issueNumber {
+			associatedIssue = issue
+			break
+		}
+	}
+	if associatedIssue == nil {
+		wh.logger.Debugf("Ignoring pull request opened event for: %s, found no repository issue with number: %d", payload.GetRepo().GetFullName(), issueNumber)
+		return
+	}
+
+	tasks, err := wh.db.GetTasks(&pb.Task{ID: associatedIssue.GetTaskID()})
+	if err != nil {
+		wh.logger.Errorf("Failed to get task from the database: %v", err)
+		return
+	}
+	if len(tasks) != 1 {
+		wh.logger.Errorf("Got an unexpected number of tasks: %d", len(tasks)) // This should never happen
+		return
+	}
+	associatedTask := tasks[0]
+
+	user, err := wh.db.GetUserByRemoteIdentity(&pb.RemoteIdentity{
+		RemoteID: uint64(payload.GetSender().GetID()),
+		Provider: "github",
+	})
+	if err != nil {
+		wh.logger.Errorf("Failed to get user from database: %v", err)
+		return
+	}
+
+	pullRequest := &pb.PullRequest{
+		ExternalRepositoryID: uint64(payload.GetRepo().GetID()),
+		TaskID:               associatedTask.GetID(),
+		IssueID:              associatedIssue.GetID(),
+		UserID:               user.GetID(),
+		SourceBranchName:     payload.GetPullRequest().GetHead().GetRef(),
+		Number:               uint64(payload.GetNumber()),
+	}
+	if !pullRequest.IsValid() {
+		wh.logger.Errorf("Failed to create pull request from event payload")
```

Comment:
Also, the IsValid check should probably be in the DB create function.
Should maybe rename it as well, since IsValid does not check for ID.
The method is more of a IsValidForCreation.

Reply:
Moved it, and renamed it to Validate

Reply:
Still not happy with this..